src/Helpers.purs

```diff
@@ -130,3 +135,20 @@ appendRightHashMap
   -> HashMap k v
   -> HashMap k v
 appendRightHashMap = HashMap.unionWith (flip const)
+
+-- | Ensures that given predicate does not hold for a value.
+never
+  :: forall (a :: Type)
+   . (a -> Boolean)
+  -> a
+  -> Maybe a
+never p a = if p a then Nothing else Just a
+
+filterMapWithKeyM
+  :: forall (m :: Type -> Type) (k :: Type) (v :: Type)
+   . Ord k
+  => Monad m
+  => (k -> v -> m Boolean)
+  -> Map k v
+  -> m (Map k v)
+filterMapWithKeyM p = map Map.fromFoldable <<< LL.filterM (uncurry p) <<< Map.toUnfoldable
```

Comment:
Just curious, why lazy list specifically?

Reply:
I have followed the implementation of `filterMapWithKey`